packages/graph/hash_graph/lib/graph/src/types/schema/mod.rs

```diff
@@ -0,0 +1,45 @@
+mod data;
+mod helper;
+pub mod properties;
+mod reference;
+
+use core::fmt;
+
+pub trait Validate {
+    /// Validates the schema of the object.
+    ///
+    /// This does only check the validity of this object, not it's child data. A validation also
+    /// happens when the type is created.
```

Comment:
I can see this being a bit misleading. We are checking if a schema is valid here. But we will also be using schemas to validate data.

Reply:
`Validate` was not supposed to be `pub` (changed this in 44805915384ac1874a8d27a5f7711b99b76e2ad5). It's used to semantically checks the schema (e.g. are all items in `"required"` also in `"properties"`) Clarified this in 678d27d197c26da7a7092f09bb4caa237301a130